src/vm/types.rs

```diff
@@ -264,24 +276,24 @@ impl TupleTypeSignature {
         for (name, type_signature) in self.type_map.iter() {
             // we only accept ascii names, so 1 char = 1 byte.
             name_size = name_size.checked_add(name.len() as i128)
-                .ok_or(Error::ValueTooLarge)?;
+                .ok_or(Error::new(ErrType::ValueTooLarge))?;
             value_size = value_size.checked_add(type_signature.size()? as i128)
-                .ok_or(Error::ValueTooLarge)?;
+                .ok_or(Error::new(ErrType::ValueTooLarge))?;
         }
         let name_total_size = name_size.checked_mul(2)
-            .ok_or(Error::ValueTooLarge)?;
+            .ok_or(Error::new(ErrType::ValueTooLarge))?;
         value_size.checked_add(name_total_size)
-            .ok_or(Error::ValueTooLarge)
+            .ok_or(Error::new(ErrType::ValueTooLarge))
     }
 
     // NOTE: this function mutates self _even if it returns an error_.
```

Comment:
I'm not seeing where the mutation happens anymore?

Reply:
These are basically used for type construction cases like:

```
(list "ab" "adc" "e")
```

Which should be of type `(list max-length: 3, dimension: 1, atomic-type: (string max-length: 3))`, but you need to iterate over the child atoms to determine the atomic type that will fit any of the children (since their individual types are length 2, 3, and 1).